### PR TITLE
featureCounts fix - require active selection of built-in annotation

### PR DESCRIPTION
**Pull request by mblue9**
This PR is a start towards a fix for https://github.com/galaxyproject/tools-iuc/issues/1754 as reported by @jennaj

### CODE REVIEW

**Comment by mvdbeek** [comment]
I like this. The other option could have an additional validator (https://docs.galaxyproject.org/en/latest/dev/schema.html#id39), but those don't work very well in workflows.

File: tools/featurecounts/featurecounts.xml [hunk 1]
```diff
@@ -134,8 +134,8 @@
                help="The input alignment file(s) where the gene expression has to be counted. The file can have a SAM or BAM format; but ALL files must be in the same format" />
         <conditional name="anno">
             <param name="anno_select" type="select" label="Gene annotation file">
+                <option value="gtf" selected="True">GTF file</option>
```
Comment (mvdbeek) [inline tools/featurecounts/featurecounts.xml]: We need a bump here, I think, since the defaults have changed.

**Comment by mblue9** [comment]
Thanks @mvdbeek I've bumped the wrapper version and also made some more changes to try to make it more robust, and eliminate some of those opportunities for errors reported by @jennaj. I've:
- made the built-in annotation selection a data table
- added some validators
- added tests for the built-in annotation and gtfs

**Comment by mvdbeek** [comment]
Very nice, thanks a lot @mblue9 !

**Pull request merged**